packages/tools/src/utilities/segmentation/thresholdVolumeByRange.ts

```diff
@@ -37,15 +48,161 @@ function thresholdVolumeByRange(
     }
   }
 
-  const callback = ({ index, pointIJK }) => {
+  // prepare a list of volume information objects for callback functions
+  const volumeInfoList = [];
+  let baseVolumeIdx = 0;
+  for (let i = 0; i < thresholdVolumeInformation.length; i++) {
+    const { imageData, spacing, dimensions } =
+      thresholdVolumeInformation[i].volume;
+
+    const volumeSize = thresholdVolumeInformation[i].volume.scalarData.length;
+    // discover the index of the volume the segmentation data is based on
+    if (volumeSize === scalarData.length) {
+      baseVolumeIdx = i;
+    }
+
+    // prepare information used in callback functions
+    const referenceValues = imageData.getPointData().getScalars().getData();
+    const lower = thresholdVolumeInformation[i].lower;
+    const upper = thresholdVolumeInformation[i].upper;
+
+    volumeInfoList.push({
+      imageData,
+      referenceValues,
+      lower,
+      upper,
+      spacing,
+      dimensions,
+      volumeSize,
+    });
+  }
+
+  // global variables used in calbackOverlap function
+  let overlaps, total, range;
+
+  /**
+   * As we are working with volumes of different dimensions, there will be no
```

Comment:
The sentence here is wrong as it is not always different dimensions. It should be something like "as there is a chance the volumes might have different dimensions and spacing ..."

Reply:
Done